### PR TITLE
Advanced workflow examples

### PR DESCRIPTION
**Pull request by ffaristocrat**
A collection of advanced workflow examples that I've assembled that aren't currently covered.

### CODE REVIEW

File: custom_email_notifications.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+version: '2.0'  # you always need this key to specify version 2 of the mistral DSL
+workflow:
+  tasks:
+    email_job_success:
+      action: civis.scripts.python3
+      input:
+        notifications:
+          successOn: true
+          successEmailAddresses:
+            - mkeane@civisanalytics.com
```
Comment (mikesaelim) [inline custom_email_notifications.yml]: I dunno if you want to be using your own email for this, you might end up getting random success/failure emails when people copy this code to try it out.

File: task_defaults.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# This example workflow demonstrates how to specify task defaults
```
Comment (mikesaelim) [inline task_defaults.yml:1]: Can you elaborate here that task defaults are default settings applied to each task in the workflow?  I know it should be obvious but it'd be helpful to spell it out.

File: variable_blocks.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+# for an introduction to YAML.
+#
+# See the Mistral documentation, https://docs.openstack.org/mistral/pike/user/dsl_v2.html,
+# for a description of the Mistral DSL.
+
+version: '2.0'  # you always need this key to specify version 2 of the mistral DSL
+workflow:
+  # Find out your Database credential id in the credential page in platform
+  # https://platform.civisanalytics.com/credentials
+  credential_id: &credential_id 1142
```
Comment (mikesaelim) [inline variable_blocks.yml]: Maybe it's best to make these `0` like you did in a previous one, just so we don't show off a working combination of IDs.

File: with_items.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# See this website, https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html,
+# for an introduction to YAML.
+#
+# See the Mistral documentation, https://docs.openstack.org/mistral/pike/user/dsl_v2.html,
+# for a description of the Mistral DSL.
+
+version: '2.0'  # you always need this key to specify version 2 of the mistral DSL
+workflow:
+  input:
+    # The collection referenced with YAQL should be under input
```
Comment (mikesaelim) [inline with_items.yml:15]: Does it have to be?  I think you can also specify the list inline, without making it an input parameter that you can modify at runtime.  But an example like this, where it's modifiable at runtime, is also useful.

File: with_items.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      - bar
+      - spam
+
+  tasks:
+    foo_bar_spam:
+      action: civis.scripts.python3
+      with-items: thing in <% $.things %>
+      input:
+        source: |
+          print(<% $.thing%>)
```
Comment (mikesaelim) [inline with_items.yml]: space

**Review by mikesaelim — CHANGES_REQUESTED** [review]

File: with_items.yml (hunk 5 above)
Comment (ffaristocrat) [reply with_items.yml]: done

File: variable_blocks.yml (hunk 3 above)
Comment (ffaristocrat) [reply variable_blocks.yml]: Done

File: task_defaults.yml (hunk 2 above)
Comment (ffaristocrat) [reply task_defaults.yml:1]: Done

File: custom_email_notifications.yml (hunk 1 above)
Comment (ffaristocrat) [reply custom_email_notifications.yml]: Done

File: with_items.yml (hunk 4 above)
Comment (ffaristocrat) [reply with_items.yml:15]: I haven't been able to get with-items to work without putting it in input. There's probably a more complicated way to reference things specified elsewhere but I don't know what it is.

**Comment by ffaristocrat** [comment]
Take another look @mikesaelim ?

**Review by mikesaelim — APPROVED** [review]
LGTM, thanks for writing these!

**Pull request merged**